examples/nodes/torchata_nodes_basics.ipynb

```diff
@@ -0,0 +1,406 @@
+    "batched_dataset = Batcher(base_node, batch_size = 4)\n",
+    "for batch in batched_dataset:\n",
+    "    print(batch)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "092d8e5b-aeb2-4de1-9b13-8b0dc0af31ce",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[0, 1, 2, 3]\n",
+      "[4, 5, 6, 7]\n",
+      "[8, 9]\n"
+     ]
+    }
+   ],
+   "source": [
+    "# We can make `drop_last = False` to produce the last batch\n",
+    "batched_dataset = Batcher(base_node, batch_size = 4, drop_last = False)\n",
+    "for batch in batched_dataset:\n",
+    "    print(batch)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "d89e36a9-cc9c-4b55-acba-dbcaf8d53407",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "On epoch = 0\n",
+      "[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]\n",
+      "On epoch = 1\n",
+      "[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]\n"
+     ]
+    }
+   ],
+   "source": [
+    "batched_dataset = Batcher(base_node, batch_size = 10)\n",
+    "\n",
+    "# If we try to use this batcher over multiple epochs, we will need to reset it after every epoch\n",
+    "num_epochs = 2\n",
+    "\n",
+    "for epoch in range(num_epochs):\n",
+    "    print(f\"On epoch = {epoch}\")\n",
+    "    for batch in batched_dataset:\n",
+    "        print(batch)\n",
+    "    batched_dataset.reset()\n",
+    "    \n",
+    "# This is one extra step than traditional dataloader, we can actually wrap the batcher in a Loader to skip that\n",
+    "# Let's look at Loader in the next cell"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d2232823-0927-432b-9171-fb2bf38a7205",
+   "metadata": {},
+   "source": [
+    "#### Loader"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "4f8a1830-49a2-4dfd-b17b-acaca7ac2e98",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "On epoch = 0\n",
+      "[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]\n",
+      "On epoch = 1\n",
+      "[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]\n"
+     ]
+    }
+   ],
+   "source": [
+    "from torchdata.nodes import Loader\n",
+    "batched_dataset = Batcher(base_node, batch_size = 10)\n",
+    "loader = Loader(batched_dataset)\n",
+    "\n",
+    "for epoch in range(num_epochs):\n",
+    "    print(f\"On epoch = {epoch}\")\n",
+    "    for batch in loader:\n",
+    "        print(batch)\n",
+    "\n",
+    "# As you can see, we get a batch in every epoch, without even needing to reset the loader!!\n",
+    "    \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "e5384863-fa3e-450e-a8fc-e838e4d2f432",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# torchadata nodes are composable, thus, many BaseNodes type nodes can be chained together for desired transformations"
```

Comment:
This should probably be markdown, maybe use this in the cell below instead of my suggestion